src/main/kotlin/cacophony/semantic/TypeChecker.kt

```diff
@@ -347,41 +358,44 @@ private class ErrorHandler(val diagnostics: Diagnostics) {
     }
 }
 
-sealed class TypeExpr(val name: String) {
-    override fun toString(): String {
-        return name
-    }
+sealed class TypeExpr(
+    val name: String,
+) {
+    override fun toString(): String = name
 
     override fun equals(other: Any?): Boolean {
         if (other == null || other !is TypeExpr) return false
         return name == other.name
     }
 
-    override fun hashCode(): Int {
-        return name.hashCode()
-    }
-
-    // TODO: Propagate Void maybe
-    // atm Void is not something you can type in code
-    object VoidType : TypeExpr("Void") // Only for `return` and `break` statement
+    override fun hashCode(): Int = name.hashCode()
 }
 
-sealed class BuiltinType private constructor(name: String) : TypeExpr(name) {
-    object BooleanType : BuiltinType("Boolean")
+sealed class BuiltinType private constructor(
+    name: String,
+) : TypeExpr(name) {
+    object BooleanType : BuiltinType("Bool")
 
     object IntegerType : BuiltinType("Int")
 
     object UnitType : BuiltinType("Unit")
+
+    object VoidType : BuiltinType("Void")
```

Comment:
Moving Void to `BuiltinType` causes problems with subtyping in function arguments and return type. So, to make it properly work, we should discuss how subtyping interacts with overloading (and I think we decided to not do that right now). I propose to remove Void from `BuiltinType` for now.

Reply:
Done